### PR TITLE
Enhance slave status by improving data api on buildslaves/builds

### PR DESCRIPTION
**Pull request by cbocta**
First PR, WIP, link to http://trac.buildbot.net/ticket/3278

idea is to provide new data api to select builds per buildslaves.
just ut ok for now.

### CODE REVIEW

File: master/buildbot/data/builds.py [hunk 1]
```diff
@@ -109,6 +111,10 @@ def startConsuming(self, callback, options, kwargs):
             # XXX these messages are never produced
             return self.master.mq.startConsuming(callback,
                                                  ('buildrequests', str(buildrequestid), 'builds', None))
```
Comment (tardyp) [inline master/buildbot/data/builds.py]: You can remove this whole startConsuming stuff. This is not used/useful

**Comment by tardyp** [comment]
Looks good to me except for the minor startConsuming technical debt.
Please try to file the front end part in a separate PR, once this one is merged

**Comment by ssavrim** [comment]
Looks good :+1:

**Comment by tardyp** [comment]
:+1:

**Comment by bastien31** [comment]
:+1:

**Comment by sa2ajj** [comment]
@cbocta you said WIP, should we merge it or wait until you say it's OK?

**Comment by djmitche** [comment]
@cbocta do you want to do anything further here?

**Comment by cbocta** [comment]
sorry for the late answer: no to me it's ok, you can merge it

**Pull request merged**